[master] fix #1240: SessionEventListener: add pre/post flush() events

Zuplyx opened the pull request:
fix #1240 
- Added preFlushUnitOfWork/postFlushUnitOfWork events
- Removed preCommitUnitOfWork event from flush in UnitOfWorkImpl, since triggering a commit event for a flush seems wrong to me.

Zuplyx commented:
@rfelcman could you please review this? Is there anything which prevents this from being merged?

rfelcman commented:
> @rfelcman could you please review this? Is there anything which prevents this from being merged?

Sorry, but our personal capacity.

rfelcman commented:
After quick look tests in
`org.eclipse.persistence.testing.tests.events.TestSessionListener` something like 
`org.eclipse.persistence.testing.tests.events.TestSessionListener#postAcquireClientSession`
`org.eclipse.persistence.testing.tests.events.SessionEventTestCase#verify`
and
something like `org.eclipse.persistence.testing.tests.unitofwork.PreCommitUnitOfWorkTest`
but for new events.
Plus copyright year update in the class headers.

Zuplyx commented:
Thank you for your feedback. I've added the tests. I also think I've updated all copyright years in the class headers. If I missed some, could you please point me towards the concrete files?

rfelcman reviewed (approved):
LGTM

Zuplyx commented:
Hello @rfelcman, I recently tried the 4.0.3 release of eclipselink and was surprised that the new event introduced by this PR was not part of it. I thought that all PRs to the master branch would be part of the next 4.x release, but it seems this is not the case. Do I have to create another PR so these changes can be used in a 4.x release?

rfelcman commented:
> Hello @rfelcman, I recently tried the 4.0.3 release of eclipselink and was surprised that the new event introduced by this PR was not part of it. I thought that all PRs to the master branch would be part of the next 4.x release, but it seems this is not the case. Do I have to create another PR so these changes can be used in a 4.x release?

Sorry but no, 
**branch  - release  -  Jakarta EE**
master - next EclipseLink 5.x - Jakarta EE 11
4.0 - EclipseLink 4.0.3 - Jakarta EE 10
3.0 - EclipseLink 3.x - Jakarta EE 9
